docs/operations.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/python
+import sys
+from ceph_wrapper import *
+import ConfigParser, os
+import subprocess
+
+
+#As of now a shell script that can accomodate rbd map and update.
+#Since there are no python rbd map and updates wrappers around ceph.
+class ShellScriptException(Exception): 
+    def __init__(self, message = None, errors = None):
+        super(ShellScriptException, self).__init__(message)
+        self.errors = errors   
+ 
+class GlobalConfig(object):
+    # once we have a config file going, this object will parse the config file.
+    # for the time being we are going to hard code the inits.
+        
+    def __init__(self):
+        self.configfile = 'bmiconfig.cfg'
+    #add parser code here once we have a configfile/ if we decide on a Db
+    # we put the db code here.
+    def __str__(self):
+        return {'file system :' : self.fstype, \
+            'rid' : self.rid, 'pool' : self.pool,\
+                'configfile' : self.r_conf} 
+
+    def parse_config(self):
+        config = ConfigParser.SafeConfigParser()
+        try:
+            config.read(self.configfile)
+            for k, v in config.items('filesystem'):
+                if v == 'True':
+                    self.fstype = k
+            if self.fstype == 'ceph':
+                self.rid = config.get(self.fstype, 'id')
+                self.pool = config.get(self.fstype, 'pool')
+                self.r_conf = config.get(self.fstype, 'conf_file')
+                self.keyring = config.get(self.fstype, 'keyring')
+        except ConfigParser.NoOptionError, err: #which is same as 'exp as e'
+            print str(err)
+
+#Calling shell script which executes a iscsi update as we don't have 
+#rbd map in documentation.
+def call_shellscript(path, m_args):
+        arglist = []
+        arglist.append(path)
+        for arg in m_args:
+                arglist.append(arg)
+        proc = subprocess.Popen(arglist, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        return proc.communicate()
+
+#Custom Exception Class that we are going to use which is a wrapper around
+#Ceph exception classes.
+class ResponseException(object):
+    def __init__(self, e):
+        self.exception_dict = dict()
+        self.exception = e
+# Extend this dict as needed for future expceptions. this
+# ensures readeability and uniformity of errors.
+        self.exception_dict = {
+           type(rbd.ImageExists())  : 401,
+           type(CephException()) : 403,
+           type(rbd.ImageBusy()) : 409,
+           type(rbd.ImageHasSnapshots()) : 405,
+           type(rbd.ImageNotFound()) : 404,
+           type(rbd.FunctionNotSupported()) : 410,
+           type(rbd.ArgumentOutOfRange()) : 411,
+           type(ShellScriptException()) : 440,
+                                }
+        #This is to handle key error exception, this also gives us the default error
```

Comment:
There is no confusion. Python should not be producing any other exception, whatsoever. Doing this is akin to doing a try"whole program" catch "do nothing, return 500". This is wrong. The end use will never see any of the other exceptions, because they will never be generated. I strongly disagreed then. I disagree now too.